src/txn/txn.c

```diff
@@ -1884,6 +1909,16 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
     txn->prepare_count = 0;
 #endif
 
+    /* Add a 2 second wait to simulate commit transaction slowness. */
+    tsp.tv_sec = 2;
+    tsp.tv_nsec = 0;
+    __wt_timing_stress(session, WT_TIMING_STRESS_COMMIT_TRANSACTION_SLOW, &tsp);
+
+    if (!prepare) {
+        WT_ERR(__txn_check_if_stable_has_moved_ahead_commit_ts(session));
+        __wt_session_gen_enter(session, WT_GEN_TXN_COMMIT);
```

Comment:
Why we enter the generation after the check?

Reply:
@quchenhao As discussed and fixed in f08e9354669258aef7f1ed7607f4960fabd9ae1b